dotnet/samples/Concepts/ChatCompletion/Onnx_ChatCompletion.cs

```diff
@@ -105,5 +105,7 @@ public async Task ChatPromptAsync()
         reply = await kernel.InvokePromptAsync(chatPrompt.ToString());
 
         Console.WriteLine(reply);
+
+        (kernel.GetRequiredService<IChatCompletionService>() as IDisposable)?.Dispose();
```

Comment:
I'm not sure if this should be a part of example, is this something that we recommend doing? Taking into account that `IChatCompletionService` is registered in DI container, should the container handle the disposal based on selected service lifecycle during registration?

Reply:
@dmytrostruk This is a non-conventional pattern introduced by Onnx Library 0.5.0, where if you leave the application / ends it without disposing the service, a message is sent to the console similar to this:

```
OGA Error: 1 instances of struct Generators::Model were leaked.
OGA Error: 1 instances of struct Generators::Tokenizer were leaked.
    Please see the documentation for the API being used to ensure proper cleanup.
```

The two instances above are part of the service, which is `IDisposable`.

Reply:
Changed internal implementation to resolve the root cause.

Reply:
@RogerBarreto Makes sense. Based on my understanding, this problem is related to console apps/test frameworks. Taking into account that `OnnxRuntimeGenAIChatCompletionService` is registered as singleton in this example, singleton objects are disposed when service provider is disposed, and in ASP.NET applications it is disposed automatically when the application is stopped/redeployed/restarted, while in console applications/test frameworks this process should be handled manually. 

The other problem is that `(kernel.GetRequiredService<IChatCompletionService>() as IDisposable)` doesn't necessarily return `OnnxRuntimeGenAIChatCompletionService` in case you have multiple `IChatCompletionService` registered and some of them are `IDisposable`, so you may accidentally dispose a wrong service.

Main goal would be to let users know that this manual disposal should happen in specific cases only and it should be done only when a resource is not needed anymore. 

Taking into account that this code is executed within a test lifecycle, we can define `Onnx_ChatCompletion` test class as `IDisposable` and then in `Dispose` method dispose not just one service, but entire service provider, which in turn should dispose all resources it contains, since it's not going to be used anymore when the test run ends:
```csharp
public void Dispose()
{
    // Dispose IServiceProvider
    if (this._serviceProvider is IDisposable disposable)
    {
        disposable.Dispose();
    }
}
```